src/lib/libfs.js

```diff
@@ -199,7 +199,12 @@ FS.staticInit();
 
           if (parts[i] === '..') {
             current_path = PATH.dirname(current_path);
-            current = current.parent;
+            if (FS.isRoot(current)) {
+              path = current_path + '/' + parts.slice(i + 1).join('/');
+              continue linkloop;
+            } else {
+              current = current.parent;
```

Comment:
If `parts[i] === '..'` and the current not is the FS root shouldn't we be failing?  (Since there is no parent of the root?)

Reply:
This is not consistent with linux's stat. If a `..` is seen at the root, it is ignored:
```
$ stat /../lib
  File: /../lib -> usr/lib
  Size: 7         	Blocks: 0          IO Block: 4096   symbolic link
```